DNS support for WebPdL2Ork

MovementGH opened the pull request:
- Added DNS resolution support to WebPdL2Ork (which was causing the issues with connecting to tweeter, since it uses domain not IP)

Tweeter should now work 100% in the browser (including connecting to the server). You can test at https://pdl2ork.movementgaming.online/?url=L2Ork-Tweeter/L2Ork-Tweeter.pd where I have a deployed version of this build

pd-l2ork commented:
This is fantastic news, @MovementGH ! BTW, I just tried loading the Tweeter and it seems to connect but not consistently. It always shows a connection being made (although the connection box is differently colored, so something parsing the colors is not quite right, possibly these are preset integer colors that may need to be fixed). That said, none of the clients update. Then, I tried going offline once, and after that the connection worked. It is super exciting to see the data flowing both ways.

There is still a small bug where the main output is not properly initialized. When on a desktop version shift+F1 will enable first part in the offline mode and its main out will immediately come up. On web version main level stays down until you manually bring it up. Something to also investigate.

Nonetheless, this is a fantastic step forward. Thanks.

MovementGH commented:
Ah, do you mean the number being pink rather than green? Now that you mention it, I do see that. As for the volume, yeah, not sure whats going on there. Volume correctly updates when loading a session or preset, but not when switching into the tweet / connecting.

I saw you connect, were you able to hear my audio? I heard yours

pd-l2ork commented:
Yes. Was that you? Your loop and instrument sounded cool. Did you save it?


Best,

Ico
________________________________
From: MovementGH ***@***.***>
Sent: Friday, January 31, 2025 9:49 PM
To: pd-l2ork/pd-l2ork ***@***.***>
Cc: Bukvic, Ivica ***@***.***>; Comment ***@***.***>
Subject: Re: [pd-l2ork/pd-l2ork] DNS support for WebPdL2Ork (PR #71)


Ah, do you mean the number being pink rather than green? Now that you mention it, I do see that. As for the volume, yeah, not sure whats going on there. Volume correctly updates when loading a session or preset, but not when switching into the tweet / connecting.

I saw you connect, were you able to hear my audio? I heard yours

—
Reply to this email directly, view it on GitHub<https://github.com/pd-l2ork/pd-l2ork/pull/71#issuecomment-2628721517>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAIEJVJ6MODDQ73UYCT77432NQY4ZAVCNFSM6AAAAABWIYSQSOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMRYG4ZDCNJRG4>.
You are receiving this because you commented.Message ID: ***@***.***>

MovementGH commented:
It was one of the already existing presets that I just loaded to test

pd-l2ork commented:
Any chance you could look into those two remaining issues:


  1.
color discrepancy (likely preset colors)
  2.
not being to connect right away (which may be linked to the init issue associated with the volume not coming up)

Best,

Ico
________________________________
From: MovementGH ***@***.***>
Sent: Friday, January 31, 2025 10:28 PM
To: pd-l2ork/pd-l2ork ***@***.***>
Cc: Bukvic, Ivica ***@***.***>; Comment ***@***.***>
Subject: Re: [pd-l2ork/pd-l2ork] DNS support for WebPdL2Ork (PR #71)


It was one of the already existing presets that I just loaded to test

—
Reply to this email directly, view it on GitHub<https://github.com/pd-l2ork/pd-l2ork/pull/71#issuecomment-2628740378>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAIEJVO3B44BHR5PVOBRMA32NQ5OPAVCNFSM6AAAAABWIYSQSOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMRYG42DAMZXHA>.
You are receiving this because you commented.Message ID: ***@***.***>

MovementGH commented:
1. Absolutely, shouldn't be too bad
2. Will look into it and see what I find

pd-l2ork commented:
Thanks! I suspect #2 is at least in part an order of execution issue.
________________________________
From: MovementGH ***@***.***>
Sent: Friday, January 31, 2025 10:39 PM
To: pd-l2ork/pd-l2ork ***@***.***>
Cc: Bukvic, Ivica ***@***.***>; Comment ***@***.***>
Subject: Re: [pd-l2ork/pd-l2ork] DNS support for WebPdL2Ork (PR #71)


  1.  Absolutely, shouldn't be too bad
  2.  Will look into it and see what I find

—
Reply to this email directly, view it on GitHub<https://github.com/pd-l2ork/pd-l2ork/pull/71#issuecomment-2628744432>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAIEJVKUIZQLYIZA7JXRUBT2NQ6YDAVCNFSM6AAAAABWIYSQSOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMRYG42DINBTGI>.
You are receiving this because you commented.Message ID: ***@***.***>

MovementGH commented:
Figured it out, Tweeter was waiting for a flat 250ms before sending its first bit of data, and since the DNS request in WebPDL2Ork has to go through before it can connect, it was bumping up against that 250ms (explains why it was very rare for me in my testing at home, since my hosted webpdl2ork is in the same room as me, but as soon as I tested from somewhere else it happened almost every time). Changed tweeter to send the first bit of data when the socket actually connects, rather than after a set delay. Please verify that I saved tweeter in the proper way, I know some parts of it are delicate. Also fixed color.

pd-l2ork commented:
Excellent work. Should I test using your hosted WebPdL2Ork? Also, did this fix that weird out of order thing where the main out does not come up upon connection or selecting offline mode?

MovementGH commented:
Have not figured out the main out yet. Also, sure. Just kicked off a pipeline to update my deployed version, it should take 5-10 minutes to complete (then make sure to clear browser cache to get the new version). If it doesnt seem to have updated let me know and I will check on it tmrw morning

pd-l2ork commented:
I just merged this. If you could decipher that last currently known inconsistency, it may lead to another potential bug in the current WebPdL2Ork implementation (perhaps how it unfolds abstractions and/or deals with GUI-backend linking).
